internal/v2/controller/http/callback.go

```diff
@@ -0,0 +1,120 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"encoding/json"
+	"net/http"
+
+	sdkCommon "github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/v2/application"
+	edgexErr "github.com/edgexfoundry/go-mod-core-contracts/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/gorilla/mux"
+)
+
+func (c *V2HttpController) DeleteDevice(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.DeleteDevice(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2DeviceCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, sdkCommon.APIV2DeviceCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateDevice(writer http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	// TODO: change to v2dto when core services are ready.
+	var device contract.Device
```

Comment:
AddDeviceRequest  & UpdateDeviceRequest  are already defined here:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/requests/device.go#L21

See other comment about how they need to be handled separately.